unchained-commons/src/main/java/unchained/commons/utils/ReflectionUtils.java

```diff
@@ -20,6 +22,310 @@ public static Type resolveCommonAncestor(List<Type> types) {
         return types.stream().reduce(ReflectionUtils::resolveCommonAncestor).orElse(null);
     }
 
+    public static <T, S extends T> Class<?> resolveRawArgument(Class<T> type, Class<S> subType) {
+        return resolveRawArgument(resolveGenericType(type, subType), subType);
+    }
+
+    public static Class<?> resolveRawArgument(Type genericType, Class<?> subType) {
+        Class<?>[] arguments = resolveRawArguments(genericType, subType);
+        if (arguments.length != 1) {
```

Comment:
Use assertions instead of explicit if statements:

```java
assertThat(arguments.length, "arguments.length", is(1));
```

This reduces code size, facilitates testing, and improves readability.